modules/swagger-codegen/src/main/resources/ruby/swagger/request.mustache

```diff
@@ -125,93 +103,60 @@ module {{moduleName}}
-      def query_string
-        # Iterate over all params,
-        # .. removing the ones that are part of the path itself.
-        # .. stringifying values so Addressable doesn't blow up.
-        query_values = {}
-        self.params.each_pair do |key, value|
-          next if self.path.include? "{#{key}}"                                   # skip path params
-          next if value.blank? && value.class != FalseClass                       # skip empties
-          if Swagger.configuration.camelize_params
-            key = key.to_s.camelize(:lower).to_sym
-          end
-          query_values[key] = value.to_s
+        if Swagger.configuration.debug
+          Swagger.logger.debug "HTTP request body param ~BEGIN~\n#{data}\n~END~\n"
         end
 
-        # We don't want to end up with '?' as our query string
-        # if there aren't really any params
-        return "" if query_values.blank?
-
-        # Addressable requires query_values to be set after initialization..
-        qs = Addressable::URI.new
-        qs.query_values = query_values
-        qs.to_s
+        data
       end
 
       def make
-        #TODO use configuration setting to determine if debugging
-        #logger = Logger.new STDOUT
-        #logger.debug self.url
-
         request_options = {
+          :method => self.http_method,
+          :headers => self.headers,
+          :params => self.params,
           :ssl_verifypeer => Swagger.configuration.verify_ssl,
-          :headers => self.headers.stringify_keys
+          :cainfo => Swagger.configuration.ssl_ca_cert,
+          :verbose => Swagger.configuration.debug
         }
-        response = case self.http_method.to_sym
-        when :get,:GET
-          Typhoeus::Request.get(
-            self.url,
-            request_options
-          )
-
-        when :post,:POST
-          Typhoeus::Request.post(
-            self.url,
-            request_options.merge(:body => self.outgoing_body)
-          )
-
-        when :patch,:PATCH
-          Typhoeus::Request.patch(
-            self.url,
-            request_options.merge(:body => self.outgoing_body)
-          )
-
-        when :put,:PUT
-          Typhoeus::Request.put(
-            self.url,
-            request_options.merge(:body => self.outgoing_body)
-          )
-
-        when :delete,:DELETE
-          Typhoeus::Request.delete(
-            self.url,
-            request_options.merge(:body => self.outgoing_body)
-          )
+
+        if [:post, :patch, :put, :delete].include?(self.http_method)
+          request_options.update :body => self.outgoing_body
+        end
+
+        raw = Typhoeus::Request.new(self.url, request_options).run
+        @response = Response.new(raw)
+
+        if Swagger.configuration.debug
+          Swagger.logger.debug "HTTP response body ~BEGIN~\n#{@response.body}\n~END~\n"
+        end
+
+        # record as last response
+        Swagger.last_response = @response
```

Comment:
This is not thread-safe, you might want to make `Swagger.last_response=` as `Thread.current[:swagger_last_response]=`?